### PR TITLE
add pow operator for exponential calculation

### PR DESCRIPTION
**Pull request by xyuan**
add new pow operator with argument of (Scalar, Pack) for exponential calculation, which is necessary used at p3 c++ subroutines.

[P3] [BFB]

### CODE REVIEW

File: components/scream/src/share/scream_pack.hpp [hunk 1]
```diff
@@ -364,6 +364,15 @@ OnlyPack<PackType> pow (const PackType& a, const ScalarType/*&*/ b) {
   return s;
 }
 
+template <typename ScalarType, typename PackType>
+KOKKOS_INLINE_FUNCTION
+OnlyPack<PackType> pow (const ScalarType& a, const PackType/*&*/ b) {
```
Comment (jgfouca) [inline components/scream/src/share/scream_pack.hpp]: I'm not 100% sure about the lack of reference for the pack. In the code immediately above, it's the Scalar that was passed by value.

Comment (ambrad) [reply components/scream/src/share/scream_pack.hpp]: Yes, please take `b` by reference.

Please also add a unit test in components/scream/src/share/tests/pack_tests.cpp

**Comment by xyuan** [comment]
Yes, this is mistake. It has been fixed. Thanks for pointing it out.

**Comment by jgfouca** [comment]
Thanks, all that's missing now is a unit test.

**Comment by AaronDonahue** [comment]
@xyuan , could you also add labels when you have a chance?  "P3" and "BFB" would be good choices for this PR.

**Comment by xyuan** [comment]
This code is ready for review, please review it. @jgfouca @AaronDonahue

**Comment by ambrad** [comment]
If you impl `pow(Pack, Pack)`, would you then be able to make the unit test simply `test_pack_gen_bin_fn_all(pow, std::pow)`? If so, that would rm the need for a special unit test for pow as well as make another potentially useful pow overload available.

**Comment by jgfouca** [comment]
Agree with @ambrad

**Comment by xyuan** [comment]
Right, that will simplify the unit test, but I am not sure whether pow(pack, pack) is used somewhere in the code, maybe we could add it first.

**Comment by ambrad** [comment]
Right, it's not currently used. So add `pow(Pack, Pack)` as a new overload, then try to use the existing `test_pack_gen_bin_fn_all` all macro for `pow`, which will test all three overloads.

**Comment by jgfouca** [comment]
Looks good

**Comment by jgfouca** [comment]
I'm a little concerned that the autotester is failing for this PR but working for a PR I issued today. @xyuan , can you do an upstream merge to confirm the failure is not simply due to your branch being out-of-date?

**Comment by xyuan** [comment]
OK, I will do it now. This PR shouldn’t be an issue since it passed the test on summit.

**Comment by jgfouca** [comment]
@xyuan did you intend to merge this PR?

**Comment by xyuan** [comment]
I already did the merge, please check. I am going to commit the p3 function conversion code if this PR is ok.

**Comment by jgfouca** [comment]
@xyuan , the autotester was telling us (correctly) that there were problems with this PR. The SCREAM dashboard now has fails for FPE testing.

**Comment by xyuan** [comment]
which compiler is used in autotester? I did the upstream merge to my branch, and test it with llvm and gcc, both are ok. How could I check the autotester error message?

**Comment by jgfouca** [comment]
@xyuan , the issue is that scream is highly sensitive to the underlying architecture, so testing on a single machine is not adequate; that's what the autotester is for.

**Comment by xyuan** [comment]
Aha, I see, should I use the same compiler as autotester and try to reproduce the error message?

**Comment by jgfouca** [comment]
@xyuan , the autotester uses different compilers for different machines, but the key thing is that different machines have different underlying architectures (CPU vs. GPU).

**Comment by ambrad** [comment]
You might check that you're building and running at least one config with the Intel compiler and SCREAM_FPE=On.

**Comment by xyuan** [comment]
Ok, I will do it now, and I will fix this PR

**Comment by jgfouca** [comment]
OK, i'm looking at it too on blake FWIW in case you don't have easy access to a machine that supports FPE.

**Comment by xyuan** [comment]
Thanks very much.

**Pull request merged**